### DISCUSSION TITLE
extracting filter state key from the URL when superset dashboard is embedded

### DISCUSSION THREAD

**Discussion by reedasha** [apache/superset#25634] [Q&A / Help] [answered]
Hi,

Is there a way to extract the filter state key either from the url or any other way when superset is embedded via an iframe into react application? I need the key to then send an API request to superset and query the value of the filter.

Thanks

**Comment by vnayagam** [comment]
Hi, did you find the solution, this is exactly where I got struck now, could you please help me ?

**Reply by reedasha** [reply, marked as answer]
I didn't find any solution on how to pull the filter state key, how I implemented it is using messaging API between iframe and react application - https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage
That required customizing superset source code and deploying our own custom image of Superset with the few lines of code that would send the filter objects up to the parent React application and then listening to the messages on the React side.

**Reply by vnayagam** [reply]
Thanks for letting me know.
